pages/project-version/middleware/index.js

```diff
@@ -196,18 +193,13 @@ const getChangedValues = (question, req) => {
     getGrantedVersion(req)
   ])
     .then(([previousVersion, grantedVersion]) => {
-      const current = getNode(req.version.data, question);
       const previous = previousVersion && getNode(previousVersion.data, question);
       const granted = grantedVersion && getNode(grantedVersion.data, question);
-
-      if (previousVersion.status === 'granted') {
-        return {
-          granted: !isEqual(current, previous) && (previous || null)
-        };
-      }
       return {
-        previous: !isEqual(current, previous) && (previous || null),
-        granted: !isEqual(current, granted) && (granted || null)
+        grantedId: grantedVersion.id,
+        previousId: previousVersion.id,
+        previous,
+        granted
```

Comment:
Also not sending the values when they hadn't changed was resulting in weirdness in the comparison window because it just assumed that the previous value was in fact `null`, and so rendered "No answer provided" for the before state, and not the actual value.